dozer-sql/src/pipeline/aggregation/avg.rs

```diff
@@ -67,93 +78,79 @@ impl Aggregator for AvgAggregator {
     }
 
     fn delete(&mut self, old: &[Field]) -> Result<Field, PipelineError> {
-        update_map(old, 1_u64, true, &mut self.current_state);
-        get_average(&self.current_state, self.return_type)
+        self.current_count -= 1;
+        get_average(
+            old,
+            &mut self.current_state,
+            &mut self.current_count,
+            self.return_type,
+            true,
+        )
     }
 
     fn insert(&mut self, new: &[Field]) -> Result<Field, PipelineError> {
-        update_map(new, 1_u64, false, &mut self.current_state);
-        get_average(&self.current_state, self.return_type)
+        self.current_count += 1;
+        get_average(
+            new,
+            &mut self.current_state,
+            &mut self.current_count,
+            self.return_type,
+            false,
+        )
     }
 }
 
 fn get_average(
-    field_map: &BTreeMap<Field, u64>,
+    field: &[Field],
+    current_sum: &mut SumState,
+    current_count: &mut u64,
     return_type: Option<FieldType>,
+    decr: bool,
 ) -> Result<Field, PipelineError> {
+    let sum = get_sum(field, current_sum, return_type, decr)?;
+
     match return_type {
         Some(FieldType::UInt) => {
-            if field_map.is_empty() {
-                Ok(Field::Decimal(calculate_err_type!(
-                    Decimal::from_f64(0_f64),
-                    Avg,
-                    FieldType::Decimal
-                )))
-            } else {
-                let mut sum =
-                    calculate_err_type!(Decimal::from_f64(0_f64), Avg, FieldType::Decimal);
-                let mut count =
-                    calculate_err_type!(Decimal::from_f64(0_f64), Avg, FieldType::Decimal);
-                for (field, cnt) in field_map {
-                    let cnt = calculate_err_field!(Decimal::from_u64(*cnt), Avg, field);
-                    sum += calculate_err_field!(field.to_decimal(), Avg, field) * cnt;
-                    count += cnt;
-                }
-                Ok(Field::Decimal(sum / count))
+            if *current_count == 0 {
+                return Ok(Field::UInt(0));
```

Comment:
`Null` is better?

Reply:
@snork-alt @mediuminvader feedback please!

Reply:
yea I think it's right to change to `Field::Null`